src/condition.h

```diff
@@ -0,0 +1,36 @@
+#ifndef CONDITION_H_
+#define CONDITION_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <Rinternals.h>
+
+static SEXP signal_condition(const char * msg, const char * class_, SEXP env = R_GlobalEnv) {
+  SEXP condition, c, signalConditionFun, out;
```

Comment:
I was being strictly c89 compliant in this header, which requires variable declarations at the top of a block.